### PR TITLE
Fixed the name attribute of file tag when using checkstyle format and no cache (2nd attempt)

### LINKED ISSUE #6097: The checkstyle format includes the current directory
## Bug report

PHP
```
PHP 7.4.25 (cli) (built: Oct 23 2021 18:05:05) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.25, Copyright (c), by Zend Technologies
```

php-cs-fixer
```
PHP CS Fixer 3.2.1 Mountains by Fabien Potencier and Dariusz Ruminski
```

I use the `--format=checkstyle` option as I am running the output through `cs2pr` in a GH Action. However `cs2pr` fails to add annotation because the file name in the XML out[ut contains the current directory.

I fixed my GH Action to remove it but I consider this a bug in php-cs-fixer.

To reproduce I have used this file

```php
<?php

class Foo
{

}
```

Run the following command 

```
php-cs-fixer fix --dry-run --using-cache=no --format=checkstyle -vvv ./test.php
```

and the output is

```xml
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle>
  <file name="CareLineLive/test.php">
    <error severity="warning" source="PHP-CS-Fixer.method_argument_space" message="Found violation(s) of type: method_argument_space"/>
  </file>
</checkstyle>
```

I believe the `name` attribute should be `name="test.php"` or maybe `name="./test.php"`.

### PR DESCRIPTION
**Pull request by troccoli**
Closes #6097

Second attempt, this time targeting master rather than 2.19 branch

### CODE REVIEW

File: tests/Console/ConfigurationResolverTest.php [hunk 1]
```diff
@@ -1170,9 +1170,15 @@ public function provideDeprecatedFixerConfiguredCases(): array
...
-            [null, '/my/path/my/file', 'path/my/file'],
+            [null, '/my/path/my/file', 'my/file'],
+            [null, './my/file', 'my/file'],
+            [null, 'my/file', 'my/file'],
             ['/my/path2/dir/.php-cs-fixer.cache', '/my/path2/dir/dir2/file', 'dir2/file'],
+            ['/my/path2/dir/.php-cs-fixer.cache', './dir2/file', 'dir2/file'],
+            ['/my/path2/dir/.php-cs-fixer.cache', 'dir2/file', 'dir2/file'],
             ['dir/.php-cs-fixer.cache', '/my/path/dir/dir3/file', 'dir3/file'],
+            ['dir/.php-cs-fixer.cache', './dir3/file', 'dir3/file'],
+            ['dir/.php-cs-fixer.cache', 'dir3/file', 'dir3/file'],
```
Comment (troccoli) [inline tests/Console/ConfigurationResolverTest.php:1181]: @keradus 

Following your suggestion I added some more test cases. If the path starts with a dot I think that should be removed from the relative path, hence why with `./my/file` I am expecting `my/file`.

File: tests/Smoke/StdinTest.php [hunk 2]
```diff
@@ -56,7 +56,7 @@ public function testFixingStdin(): void
         $fileResult = str_replace("\n--- ".$path."\n", "\n--- php://stdin\n", $fileResult);
         $fileResult = str_replace("\n+++ ".$path."\n", "\n+++ php://stdin\n", $fileResult);
 
-        $path = str_replace('/', \DIRECTORY_SEPARATOR, basename(realpath($cwd)).'/'.$inputFile);
+        $path = str_replace('/', \DIRECTORY_SEPARATOR, $inputFile);
```
Comment (troccoli) [inline tests/Smoke/StdinTest.php:59]: @keradus 

In the previous PR you said
> isn't this change braking the usage of fixing the relatives and symlink/linked files?
>
> maybe it's worth to add a test cases like those

I'm not sure I understand what you mean. This is a test and the change adjust what we expect from stdout.

**Comment by CLL-GTA** [comment]
@keradus 

Something else I could do for this PR to be merged?

**Comment by Wirone** [comment]
Thank you very much for taking your time and working on this, also sorry for lack of response 😞. Good news: closing as this is not reproducible on `3.16.1-DEV` (current master with unreleased changes):

image

Most probably fixed in #6915.

**Comment by CLL-GTA** [comment]
Awesome. Thanks.

**Pull request closed without merging**